docs/src/components/DocLayout/index.tsx

```diff
@@ -0,0 +1,160 @@
+import GuidelineImages, { DoImage, DontImage } from "../Guidelines/GuidelineImages";
+import HeaderLink from "../HeaderLink";
+import ImageContainer from "../ImageContainer";
+import Navbar from "../Navbar";
+import { BookmarkProvider } from "../../services/bookmarks";
+import DocNavigation from "../DocNavigation";
+import Breadcrumbs from "../Breadcrumbs";
+import ComponentStatus from "../ComponentStatus";
+import TableOfContents, { TocItemObject } from "../TableOfContents";
+import Tabs, { TabObject } from "../Tabs";
+import ReactExample from "../ReactExample";
+import Footer from "../Footer";
+import StyledWrapper from "./primitives/StyledWrapper";
+import StyledMiddle from "./primitives/StyledMiddle";
+import StyledMain from "./primitives/StyledMain";
+import StyledMobileOutdent from "./primitives/StyledMobileOutdent";
+import StyledTocWrapper from "./primitives/StyledTocWrapper";
+import StyledProse from "./primitives/StyledProse";
+
+interface Props {
+  children: React.ReactNode;
+  description?: string;
+  headerLink?: string;
+  location: WindowLocation;
+  noElevation?: boolean;
+  path: string;
+  tableOfContents: TocItemObject[];
+  tabs?: TabObject[];
+  title?: string;
+  trail?: Array<{
+    name: string;
+    url: string;
+  }>;
+}
+
+export default function DocLayout({
+  children,
+  description,
+  headerLink,
+  location,
+  noElevation,
+  path,
+  tableOfContents,
+  tabs,
+  title,
+  trail,
+}: Props) {
+  const Toc = <TableOfContents items={tableOfContents} />;
+  const tocHasItems = tableOfContents?.length > 0;
+  const navigation = <DocNavigation currentUrl={path} />;
+  return (
+    <ThemeProvider theme={defaultTheme}>
+      <DevModeProvider>
+        <BookmarkProvider page={path} location={location}>
+          <StyledWrapper>
+            <Navbar location={location} docNavigation={navigation} />
+            <StyledMiddle>
+              <Stack
+                flex
+                direction="column"
+                desktop={{ direction: "row", spacing: "large", align: "stretch" }}
+              >
+                {trail && (
+                  <Hide block on={["smallMobile", "mediumMobile", "largeMobile", "tablet"]}>
+                    {navigation}
+                  </Hide>
+                )}
+                <StyledMain>
+                  {trail && <Breadcrumbs trail={trail} />}
+                  <Box padding={{ bottom: "XLarge" }}>
+                    <Stack inline align="center" spaceAfter="small">
+                      <AddBookmark />
+                      <Heading as="h1" type="display">
+                        {title}
+                      </Heading>
+                    </Stack>
+                    {description && (
+                      <Box padding={{ left: "XXLarge" }}>
+                        <Text>{description}</Text>
+                      </Box>
+                    )}
+                  </Box>
+                  <StyledMobileOutdent>
```

Comment:
here is a visual bug, when the icon is not placed on the bottom
![Screenshot 2021-04-30 at 13 20 47](https://user-images.githubusercontent.com/14054752/116688388-ea997680-a9b6-11eb-99d6-7a729ab0ee37.png)